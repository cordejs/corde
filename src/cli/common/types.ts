import { Command } from "commander";

export type ParamsFrom = "args" | "options" | "both";

export interface ICommandOptions {
  flags: string;
  description?: string;
  defaultValue?: string;
}

export interface ICliCommandConstructor {
  name?: string;
  program: Command;
  paramsFrom?: ParamsFrom;
}

export interface ICliCommand {
  readonly paramsFrom?: ParamsFrom;
  readonly command: Command;
  handler(...args: any[]): Promise<void> | void;
}

export interface ICliCommandConstructable {
  new (command: Command): ICliCommand;
}
