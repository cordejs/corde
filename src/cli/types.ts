export interface ICommand {
  action(...args: any[]): Promise<void> | void;
}
