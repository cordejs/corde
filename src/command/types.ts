import { ICordeBot } from "../types";

export interface ICreateMatcherParam {
  matcher?: string;
  isNot?: boolean;
  commandName?: string | boolean | number;
  channelId?: string;
  guildId?: string;
  isDebug?: boolean;
  isCascade?: boolean;
  cordeBot?: ICordeBot;
  trace?: string;
  utility?: boolean;
  mustSendCommand?: boolean;
}
