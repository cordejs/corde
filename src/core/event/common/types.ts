import { Client, ClientEvents } from "discord.js";

export type EventNames = keyof ClientEvents;

export type Spread<T extends EventNames> = ClientEvents[T][1] extends undefined
  ? ClientEvents[T] extends (infer U)[]
    ? U
    : never
  : ClientEvents[T];

export interface IDiscordEvent<TEvent extends EventNames, TFilter> {
  canListen(): boolean;
  getIntents(): string[];
  on(fn: (...args: ClientEvents[TEvent]) => void): this;
  once(filter?: TFilter): Promise<Spread<TEvent>>;
}

export interface IDiscordEventConstructable<TEvent extends EventNames, TFilter> {
  new (client: Client): IDiscordEvent<TEvent, TFilter>;
}

export type IntentType =
  | "GUILDS"
  | "GUILD_MEMBERS"
  | "GUILD_BANS"
  | "GUILD_EMOJIS_AND_STICKERS"
  | "GUILD_INTEGRATIONS"
  | "GUILD_WEBHOOKS"
  | "GUILD_INVITES"
  | "GUILD_VOICE_STATES"
  | "GUILD_PRESENCES"
  | "GUILD_MESSAGES"
  | "GUILD_MESSAGE_REACTIONS"
  | "GUILD_MESSAGE_TYPING"
  | "DIRECT_MESSAGES"
  | "DIRECT_MESSAGE_REACTIONS"
  | "DIRECT_MESSAGE_TYPING"
  | "GUILD_SCHEDULED_EVENTS";

export type EventClass = {
  [k in IntentType]: {
    bitfield: number;
    events: EventNames[];
  };
};
