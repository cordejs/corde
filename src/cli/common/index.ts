export * from "./loadConfigs";
export * from "./commandFactory";
