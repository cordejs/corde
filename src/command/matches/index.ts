export * from "./message";
export * from "./role";
