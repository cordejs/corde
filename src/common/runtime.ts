import { IConfigOptions, ICordeBot, TestFunctionType } from "../types";
import { Config } from "./config";
import { Client } from "discord.js";
import { CordeBot } from "../core/cordeBot";
import { ConfigError } from "../errors";

const Environment = {
  isUnityTest: process.env.ENV === "UNITY_TEST",
  isE2eTest: process.env.ENV === "E2E_TEST",
};

/**
 * @internal
 */
class Runtime implements IConfigOptions {
  get bot() {
    if (!this._bot) {
      this._bot = this.initBot();
    }

    return this._bot;
  }

  get isTestEnv() {
    return this.environment.isE2eTest || this.environment.isUnityTest;
  }

  get environment() {
    return Environment;
  }

  get exitOnFileReadingError() {
    return this._configs.exitOnFileReadingError;
  }

  get extensions() {
    return this._configs.extentions;
  }

  get events() {
    return this.bot.events;
  }

  get configs() {
    return this._configs;
  }

  get cordeBotToken() {
    return this._configs.cordeBotToken;
  }

  get botTestId() {
    return this._configs.botTestId;
  }

  get project() {
    return this._configs.project;
  }

  get botToken() {
    return this._configs.botToken;
  }

  get channelId() {
    return this._configs.channelId;
  }

  get guildId() {
    return this._configs.guildId;
  }

  get timeout() {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    return this._configs.timeout!;
  }

  get botPrefix() {
    return this._configs.botPrefix;
  }

  get testMatches() {
    return this._configs.testMatches;
  }

  get modulePathIgnorePatterns() {
    return this._configs.modulePathIgnorePatterns;
  }

  private constructor() {
    this._configs = new Config();
  }

  private static _instance: Runtime;
  configFilePath!: string;
  files!: string[];

  private readonly _configs: Config;
  private _bot!: ICordeBot;

  static getInstance() {
    if (!Runtime._instance) {
      Runtime._instance = new Runtime();
    }
    return Runtime._instance;
  }

  setConfigs(_configs: Partial<IConfigOptions>, forceUpdate?: boolean) {
    if (!_configs) {
      throw new ConfigError("Invalid _configs");
    }

    this._configs.setConfigs(_configs, forceUpdate);
  }

  /**
   * Shortcut for *bot.isLoggedIn*
   */
  isBotLoggedIn() {
    return this.bot && this.bot.isLoggedIn();
  }

  /**
   * Shortcut for *bot.logout*
   */
  logoffBot() {
    if (this._bot) {
      this._bot.logout();
    }
  }

  async loginBot(token: string) {
    return await this.bot.login(token);
  }

  injectBot(fn: TestFunctionType) {
    return fn(this.bot);
  }

  initBot() {
    return new CordeBot(
      this._configs.botPrefix,
      this._configs.guildId,
      this._configs.channelId,
      this._configs.botTestId,
      new Client(),
    );
  }
}

/**
 * Singleton of Runtime.
 */
const runtime = Runtime.getInstance();
export { runtime };
