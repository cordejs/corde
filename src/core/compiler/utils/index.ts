export * from "./import-js-in-directory";
export * from "./was-file-modified";
