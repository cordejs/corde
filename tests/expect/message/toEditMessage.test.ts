import { Client } from "discord.js";
import MockDiscord from "../../mocks/mockDiscord";
import { createCordeBotWithMockedFunctions, testUtils } from "../../testHelper";
import { ICordeBot, IMessageEmbed, ITestReport } from "../../../src/types";
import { ToEditMessage } from "../../../src/expect/matches";
import { buildReportMessage, diff, formatObject } from "../../../src/utils";
import { runtime } from "../../../src/common/runtime";
import { MockEvents } from "../../mocks/mockEvents";

let mockDiscord = new MockDiscord();

describe("testing toEditMessage", () => {
  afterEach(() => {
    mockDiscord = new MockDiscord();
  });

  function initTestClass(cordeBot: ICordeBot, commandName?: string, isNot?: boolean) {
    return testUtils.initTestClass(ToEditMessage, {
      command: commandName,
      cordeBot: cordeBot,
      isNot: isNot,
      channelId: runtime.channelId,
    });
  }

  it("should return a failed test due to invalid parameter (null)", async () => {
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const message = buildReportMessage(
      `expected: expect value to be a primitive value (string, boolean, number) or an IMessageEmbed object\n`,
      `received: null`,
    );

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      message,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(null);
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should return a failed test due to invalid parameter (undefined)", async () => {
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const message = buildReportMessage(
      `expected: expect value to be a primitive value (string, boolean, number) or an IMessageEmbed object\n`,
      `received: undefined`,
    );

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      message,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(undefined);
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should fail due to message was not edited by the bot (no messageIdentifier)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const message = buildReportMessage(
      `expected: testing bot to edit the last message sent\n`,
      `received: message was not edited`,
    );

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      message,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action("pong");
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should fail due to message was not edited by the bot (no messageIdentifier as string)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const message = buildReportMessage(
      `expected: testing bot to edit the message of id 123\n`,
      `received: message was not edited`,
    );

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      message,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action("pong", "123");
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should fail due to message was not edited by the bot (no messageIdentifier as object with id)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const message = buildReportMessage(
      `expected: testing bot to edit the message of id 123\n`,
      `received: message was not edited`,
    );

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      testName: toEditMessage.toString(),
      message,
    };

    const report = await toEditMessage.action("pong", { id: "123" });
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should fail due to message was not edited by the bot (no messageIdentifier as object with content)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const message = buildReportMessage(
      `expected: testing bot to edit the message of content "message test"\n`,
      `received: message was not edited`,
    );

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      message,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action("pong", { oldContent: "message test" });
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get success test due to timeout but isNot = true", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const toEditMessage = initTestClass(cordeClient, "ping", true);

    const reportModel: ITestReport = {
      pass: true,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action("pong");
    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get success test due to bot returned equal message", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange();

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: true,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(mockDiscord.message.content);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get success test due to bot returned equal messages (string type)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange();

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: true,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(mockDiscord.message.content);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get success test due to bot returned equal messages (string number)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());
    mockDiscord.message.content = "2";

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange();

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: true,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(2);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get success test due to bot returned equal messages (type embed)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange(mockDiscord.messageWithEmbed);

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: true,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(mockDiscord.messageEmbedSimple);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get failed test due to bot returned equal messages (isNot true)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange(mockDiscord.messageWithEmbed);

    const toEditMessage = initTestClass(cordeClient, "ping", true);

    const reportModel: ITestReport = {
      pass: false,
      testName: toEditMessage.toString(),
      message: buildReportMessage(
        `expected: message from bot be different from expectation\n`,
        `received: both returned and expectation are equal`,
      ),
    };

    const report = await toEditMessage.action(mockDiscord.messageEmbedSimple);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get success test due to bot returned different messages (isNot true)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange();

    const toEditMessage = initTestClass(cordeClient, "ping", true);

    const reportModel: ITestReport = {
      pass: true,
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(mockDiscord.messageEmbedSimple);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get fail test due to bot returned different messages (both embed)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange(mockDiscord.messageWithEmbed);

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const embedReturned = toEditMessage.messageEmbedToMessageEmbedInterface(
      mockDiscord.messageWithEmbed.embeds[0],
    );

    const embedInternal: IMessageEmbed = {
      fields: mockDiscord.messageEmbedSimple.fields,
    };

    const reportModel: ITestReport = {
      pass: false,
      testName: toEditMessage.toString(),
      message: buildReportMessage(diff(embedReturned, embedInternal)),
    };

    const report = await toEditMessage.action(embedInternal);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get fail test due to bot returned different messages (expect primitive and returned embed)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange(mockDiscord.messageWithEmbed);
    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const expectValue = "expect value";
    const embedReturned = toEditMessage.getMessageByType(mockDiscord.messageWithEmbed, "embed");

    const reportModel: ITestReport = {
      pass: false,
      testName: toEditMessage.toString(),
      message: buildReportMessage(
        `expected: '${expectValue}'\n`,
        `received: ${formatObject(embedReturned)}`,
      ),
    };

    const report = await toEditMessage.action(expectValue);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get fail test due to bot returned different messages both primitive values", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());
    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange();

    const expectValue = "expect value";

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const reportModel: ITestReport = {
      pass: false,
      message: buildReportMessage(
        `expected: '${expectValue}'\n`,
        `received: '${mockDiscord.message.content}'`,
      ),
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(expectValue);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should get fail test due to bot returned different messages (expect embed and returned primitive)", async () => {
    runtime.setConfigs({ timeout: 100 }, true);
    const cordeClient = createCordeBotWithMockedFunctions(mockDiscord, new Client());

    const mockEvent = new MockEvents(cordeClient, mockDiscord);
    mockEvent.mockOnceMessageContentOrEmbedChange();

    const toEditMessage = initTestClass(cordeClient, "ping", false);

    const events = new MockEvents(cordeClient, mockDiscord);
    events.mockOnceMessageContentOrEmbedChange();

    const reportModel: ITestReport = {
      pass: false,
      message: buildReportMessage(
        `expected: ${formatObject(mockDiscord.messageEmbedSimple)}\n`,
        `received: '${mockDiscord.message.content}'`,
      ),
      testName: toEditMessage.toString(),
    };

    const report = await toEditMessage.action(mockDiscord.messageEmbedSimple);

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });

  it("should return a failed test due to failure in message sending", async () => {
    const corde = createCordeBotWithMockedFunctions(mockDiscord, new Client());
    corde.getRoles = jest.fn().mockReturnValue(mockDiscord.roleManager.cache);
    corde.findRole = jest.fn().mockReturnValue(mockDiscord.role);

    const erroMessage = "can not send message to channel x";
    corde.sendTextMessage = jest
      .fn()
      .mockImplementation(() => Promise.reject(new Error(erroMessage)));

    const toEditMessage = initTestClass(corde, "");
    const report = await toEditMessage.action("value", { id: "123" });

    const reportModel: ITestReport = {
      pass: false,
      message: buildReportMessage(erroMessage),
      testName: toEditMessage.toString(),
    };

    expect(report).toEqual(reportModel);
    expect(report).toMatchSnapshot();
  });
});
