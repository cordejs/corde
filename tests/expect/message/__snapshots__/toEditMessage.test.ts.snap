// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing toEditMessage should fail due to message was not edited by the bot (no messageIdentifier as object with content) 1`] = `
Object {
  "message": "    
    expected: testing bot to edit the message of content \\"message test\\"
    received: message was not edited
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should fail due to message was not edited by the bot (no messageIdentifier as object with id) 1`] = `
Object {
  "message": "    
    expected: testing bot to edit the message of id 123
    received: message was not edited
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should fail due to message was not edited by the bot (no messageIdentifier as string) 1`] = `
Object {
  "message": "    
    expected: testing bot to edit the message of id 123
    received: message was not edited
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should fail due to message was not edited by the bot (no messageIdentifier) 1`] = `
Object {
  "message": "    
    expected: testing bot to edit the last message sent
    received: message was not edited
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get fail test due to bot returned different messages (both embed) 1`] = `
Object {
  "message": "    
    [32m- expected  - 10[39m
    [31m+ received  +  8[39m
    
    [2m  Object {[22m
    [2m    \\"author\\": Object {[22m
    [32m-     \\"icon_url\\": \\"https://i.imgur.com/wSTFkRM.png\\",[39m
    [32m-     \\"name\\": \\"Some name\\",[39m
    [32m-     \\"url\\": \\"https://discord.js.org\\",[39m
    [31m+     \\"icon_url\\": undefined,[39m
    [31m+     \\"name\\": \\"ITest\\",[39m
    [31m+     \\"url\\": undefined,[39m
    [2m    },[22m
    [32m-   \\"color\\": 39423,[39m
    [32m-   \\"description\\": \\"Some description here\\",[39m
    [31m+   \\"color\\": null,[39m
    [31m+   \\"description\\": null,[39m
    [2m    \\"fields\\": Array [[22m
    [2m      Object {[22m
    [2m        \\"inline\\": false,[22m
    [2m        \\"name\\": \\"Regular field title\\",[22m
    [2m        \\"value\\": \\"Some value here\\",[22m
    [2m      },[22m
    [2m      Object {[22m
    [2m        \\"inline\\": false,[22m
    [2m        \\"name\\": \\"â€‹\\",[22m
    [2m        \\"value\\": \\"â€‹\\",[22m
    [2m      },[22m
    [2m      Object {[22m
    [2m        \\"inline\\": true,[22m
    [2m        \\"name\\": \\"Inline field title\\",[22m
    [2m        \\"value\\": \\"Some value here\\",[22m
    [2m      },[22m
    [2m      Object {[22m
    [2m        \\"inline\\": true,[22m
    [2m        \\"name\\": \\"Inline field title\\",[22m
    [2m        \\"value\\": \\"Some value here\\",[22m
    [2m      },[22m
    [2m    ],[22m
    [2m    \\"footer\\": null,[22m
    [2m    \\"image\\": null,[22m
    [32m-   \\"thumbnail\\": Object {[39m
    [32m-     \\"url\\": \\"https://i.imgur.com/wSTFkRM.png\\",[39m
    [32m-   },[39m
    [31m+   \\"thumbnail\\": null,[39m
    [2m    \\"timestamp\\": null,[22m
    [32m-   \\"title\\": \\"Some title\\",[39m
    [31m+   \\"title\\": null,[39m
    [2m    \\"type\\": \\"rich\\",[22m
    [32m-   \\"url\\": \\"https://discord.js.org/\\",[39m
    [31m+   \\"url\\": null,[39m
    [2m  }[22m
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get fail test due to bot returned different messages (expect embed and returned primitive) 1`] = `
Object {
  "message": "    
    expected: Object {
      \\"author\\": Object {
        \\"icon_url\\": \\"https://i.imgur.com/wSTFkRM.png\\",
        \\"name\\": \\"Some name\\",
        \\"url\\": \\"https://discord.js.org\\",
      },
      \\"color\\": 39423,
      \\"description\\": \\"Some description here\\",
      \\"fields\\": Array [
        Object {
          \\"inline\\": false,
          \\"name\\": \\"Regular field title\\",
          \\"value\\": \\"Some value here\\",
        },
        Object {
          \\"inline\\": false,
          \\"name\\": \\"â€‹\\",
          \\"value\\": \\"â€‹\\",
        },
        Object {
          \\"inline\\": true,
          \\"name\\": \\"Inline field title\\",
          \\"value\\": \\"Some value here\\",
        },
        Object {
          \\"inline\\": true,
          \\"name\\": \\"Inline field title\\",
          \\"value\\": \\"Some value here\\",
        },
      ],
      \\"footer\\": null,
      \\"image\\": null,
      \\"thumbnail\\": Object {
        \\"url\\": \\"https://i.imgur.com/wSTFkRM.png\\",
      },
      \\"timestamp\\": null,
      \\"title\\": \\"Some title\\",
      \\"type\\": \\"rich\\",
      \\"url\\": \\"https://discord.js.org/\\",
    }
    received: 'this is the message content'
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get fail test due to bot returned different messages (expect primitive and returned embed) 1`] = `
Object {
  "message": "    
    expected: 'expect value'
    received: Object {
      \\"author\\": Object {
        \\"icon_url\\": \\"https://i.imgur.com/wSTFkRM.png\\",
        \\"name\\": \\"Some name\\",
        \\"url\\": \\"https://discord.js.org\\",
      },
      \\"color\\": 39423,
      \\"description\\": \\"Some description here\\",
      \\"fields\\": Array [
        Object {
          \\"inline\\": false,
          \\"name\\": \\"Regular field title\\",
          \\"value\\": \\"Some value here\\",
        },
        Object {
          \\"inline\\": false,
          \\"name\\": \\"â€‹\\",
          \\"value\\": \\"â€‹\\",
        },
        Object {
          \\"inline\\": true,
          \\"name\\": \\"Inline field title\\",
          \\"value\\": \\"Some value here\\",
        },
        Object {
          \\"inline\\": true,
          \\"name\\": \\"Inline field title\\",
          \\"value\\": \\"Some value here\\",
        },
      ],
      \\"footer\\": null,
      \\"image\\": null,
      \\"thumbnail\\": Object {
        \\"url\\": \\"https://i.imgur.com/wSTFkRM.png\\",
      },
      \\"timestamp\\": null,
      \\"title\\": \\"Some title\\",
      \\"type\\": \\"rich\\",
      \\"url\\": \\"https://discord.js.org/\\",
    }
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get fail test due to bot returned different messages both primitive values 1`] = `
Object {
  "message": "    
    expected: 'expect value'
    received: 'this is the message content'
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get failed test due to bot returned equal messages (isNot true) 1`] = `
Object {
  "message": "    
    expected: message from bot be different from expectation
    received: both returned and expectation are equal
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get success test due to bot returned different messages (isNot true) 1`] = `
Object {
  "pass": true,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get success test due to bot returned equal message 1`] = `
Object {
  "pass": true,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get success test due to bot returned equal messages (string number) 1`] = `
Object {
  "pass": true,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get success test due to bot returned equal messages (string type) 1`] = `
Object {
  "pass": true,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get success test due to bot returned equal messages (type embed) 1`] = `
Object {
  "pass": true,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should get success test due to timeout but isNot = true 1`] = `
Object {
  "pass": true,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should return a failed test due to failure in message sending 1`] = `
Object {
  "message": "    
    can not send message to channel x
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should return a failed test due to invalid parameter (null) 1`] = `
Object {
  "message": "    
    expected: expect value to be a primitive value (string, boolean, number) or an IMessageEmbed object
    received: null
",
  "pass": false,
  "testName": "toEditMessage",
}
`;

exports[`testing toEditMessage should return a failed test due to invalid parameter (undefined) 1`] = `
Object {
  "message": "    
    expected: expect value to be a primitive value (string, boolean, number) or an IMessageEmbed object
    received: undefined
",
  "pass": false,
  "testName": "toEditMessage",
}
`;
